src/SharedInfrastructure/Guard.cs

```diff
@@ -90,7 +85,7 @@ public static void MustBeLessThanOrEqualTo<TValue>(TValue value, TValue max, str
         {
             if (value.CompareTo(max) > 0)
```

Comment:
I was never fan of the `IComparer<T>` solution here. It makes `Guard` useless on hot path, [check out the difference](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBDAzgWwB8ABAJgEYBYAKGIAYACY8gOgBEBLbAcwDsJcGDmFwBuGvSatOPfoOG4WAYQgATGAEFe2ADYBPXBzETGzFgCUArryH4YyiPgAOHHTCgBldwDdhMY9QSAMxMpAxKDADeNAyxDDFxxCHMSEwoDADiMLzuwgAUHDYMCACU0dRxlZmW2FCqLACyloIAQjAAMv64ACoAFti8ADyFGAB8eQhoDACsJeIVcQC+CbErTMnkqcTpAHIQvFk5UPkjxWVrlRk1dY3NGG2duD39vBNTs/OVy9TfNBJhV1qqiiawA2g0YBhemoAJLOHR5CFQ2HwgDyTiE+0UGm43FgTw43hgMN4OkKhW4JQAumsklItukmq0Ol0+gNBt0AGq6SwwcZcnkwBjeQVTAU6XkMfDYUoXKqxADuvXcQvFkpADBhKmctWwwDcHO5Er5cti5Xl8o4ADMGHkRcaHDrYN0IHlpaUGKMALwMOglU1Vc0W4N9KAQBV5OYBr4B76VMFI6GqOFOBGJlGp9GY3jY3H4wxEklk3gU6m0jYMhhM+4sp5s16ne28qand3+hbyoPB2LW21NoXeqUy9vduJd0dxUPhyOfCdxi3z1Yd+LLhOQpMptPrjM6LMcLEsPZF8m8Sk05dOY4ijBClJpBhTiMlBhe0YMKFhhUMHJfgCiCDAGAMX3V4ACIQDyUCox+GggA)!

Can we replace add non-generic overloads at least for most common types like `(U)Int32/64`? Or maybe completely replace the generic solution with the specific implementations, if you have time.

Reply:
**ARGH** I hadn't noticed the two conditional branches in the `int.CompareTo` method, ew!

![image](https://user-images.githubusercontent.com/10199417/75594790-3f2b2400-5a8a-11ea-8fe1-08c43fcbdbd2.png)

Yeah I can definitely add some overloads for `int`/`long`, sure 👍
I would say though we could still leave the generic overload just in case we need it in the future?

Reply:
We don't need the generic variants, if we have overloads for all the [integral types](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/builtin-types/integral-numeric-types) + `float` + `double`. If you do it only for `int` + `long`, then keep the generic ones.

Maybe T4 can help?

Reply:
Overloads for `int`/`long` done in https://github.com/SixLabors/SharedInfrastructure/pull/7/commits/9eccb6a09ae34870cd66a0c7e8a3613ec79d0f62 👍